numpy/core/src/multiarray/dtypemeta.c

```diff
@@ -208,6 +219,34 @@ flexible_default_descr(PyArray_DTypeMeta *cls)
 }
 
 
+static PyArray_Descr *
+string_unicode_common_instance(PyArray_Descr *descr1, PyArray_Descr *descr2)
+{
+    if (descr1->elsize >= descr2->elsize) {
```

Comment:
I changed this to prefer the first dtype. Doesn't change anything except which identity gets returned, which also means the "metadata" that is returned is different.

This is a tiny change here: Since the other dtype gets cast first and then we call "common instance", the metadata and instance which is returned can differ from the one that was returned before.